src/domain/collection/CollectionPageContainer.tsx

```diff
@@ -35,50 +39,78 @@ const CollectionPageContainer: React.FC = () => {
   });
   const collection = collectionData && collectionData.collectionDetails;
 
+  const getErrorHero = (collection?: CollectionFieldsFragment) => {
+    if (!collection) {
+      return (
+        <ErrorHero
+          text={t('collection.notFound.text')}
+          title={t('collection.notFound.title')}
+        >
+          <Link to={`/${locale}${ROUTES.EVENTS}${search}`}>
+            {t('collection.notFound.linkSearchEvents')}
+          </Link>
+        </ErrorHero>
+      );
+    }
+
+    if (!isLanguageSupported(collection, locale)) {
+      return (
+        <ErrorHero
+          text={t('collection.languageNotSupported.text')}
+          title={t('collection.languageNotSupported.title')}
+        >
+          <Link to={`/${locale}${ROUTES.COLLECTIONS}`}>
+            {t('collection.languageNotSupported.linkSearchEvents')}
+          </Link>
+        </ErrorHero>
+      );
+    }
+
+    if (isCollectionExpired(collection)) {
+      return (
+        <ErrorHero
+          text={t('collection.expired.text')}
+          title={t('collection.expired.title')}
+        >
+          <Link to={`/${locale}${ROUTES.COLLECTIONS}`}>
+            {t('collection.expired.linkSearchEvents')}
+          </Link>
+        </ErrorHero>
+      );
+    }
+  };
+
+  const errorHero = getErrorHero(collection);
+
   return (
     <PageWrapper
       className={styles.collectionPageWrapper}
       title="collection.title"
     >
       <LoadingSpinner isLoading={loading}>
-        {collection ? (
-          <>
-            {!!isLanguageSupported(collection, locale) ? (
-              <>
-                <CollectionPageMeta collection={collection} />
+        <>
+          {!collection ? (
+            errorHero
+          ) : (
+            <>
+              <CollectionPageMeta collection={collection} />
+
+              {errorHero ? (
```

Comment:
maybe 

errorHero || <></>

would improve readability?

Reply:
Refactored way to get errors little bit. I hope readability is better now